viewing_party/party.py

```diff
@@ -1,23 +1,193 @@
 # ------------- WAVE 1 --------------------
+import copy
+from collections import Counter
 
+#1
 def create_movie(title, genre, rating):
-    pass
+    movie_dic = {}
+    if not title or not genre or not rating:
+        return None      
+    else: 
+         movie_dic["title"] = title
+         movie_dic["genre"] = genre
+         movie_dic["rating"] = rating
+         return movie_dic
+#2
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+#3
+def add_to_watchlist(user_data,movie):
+   user_data["watchlist"].append(movie)
+   return user_data
+
+#4 
+def watch_movie(user_data, title):
+    user_data_copy = copy.deepcopy(user_data)
+    for movie in user_data["watchlist"]:
+        if title in movie["title"]:
+            user_data_copy["watchlist"].remove(movie)
+            user_data_copy["watched"].append(movie)
+            break

```

Comment:
Good job remembering to add `break` after modifying a mutable collection during iteration. This is something we have to be careful about. 

Alternatively, you could only use the loop to find the value we want to remove (either the entire value, if we later use `remove`, or the position, if we later use `pop`), and then remove the value after the loop completes.

Here's a [resource](https://support.techsmart.codes/hc/en-us/articles/360058068873-For-Loops-Why-shouldn-t-I-add-to-or-remove-from-a-list-during-a-for-each-loop) about why we shouldn't mutate a collection while iterating over it if you'd like to review.